chaoslib/openebs/cstor_replica_network_delay.yaml

```diff
@@ -0,0 +1,191 @@
+  register: tc_apt_output
+
+- name: Inject packet loss on targeted replica
+  shell: kubectl exec -it {{ targeted_replica_pod }} -n  openebs --container cstor-pool -- bash -c "tc qdisc add dev eth0 root netem loss 100.00"
+  register: tc_output
+
+- name: Check if targeted replica is disconnected
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n openebs   --container  cstor-istgt -- istgtcontrol -q replica |json_pp |grep "\"status" |awk -F ':' '{print $2}' | tr -d ',' |wc -l
+  register: connected_replica
+  until: "connected_replica.stdout == \"2\""
+  delay: 30
+  retries: 10
+
+#- name: Get the node on which the targeted cstor replica is scheduled
+#  shell: >
+#    kubectl get pod {{ targeted_replica_pod }} -n {{ operator_ns }}
+#    --no-headers -o custom-columns=:spec.nodeName
+#  args:
+#    executable: /bin/bash
+#  register: node
+
+#- name: Identify the pumba pod that co-exists with targeted cstor replica pod
+#  shell: >
+#    kubectl get pods -l app=pumba -n {{ app_ns }}
+#    -o jsonpath='{.items[?(@.spec.nodeName==''"{{ node.stdout }}"'')].metadata.name}'
+#  args:
+#    executable: /bin/bash
+#  register: pumba_pod
+
+- name: Get current iostats from target
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n openebs --container cstor-istgt -- istgtcontrol -q iostats |json_pp| grep WriteIOPS |awk -F ':' '{print $2}' | tr -d -c 0-9
+  register: last_iostats
+
+#- name: Inject packet loss for {{ NETWORK_DELAY }}s on targeted cstor replica
+#  shell: >
+#    kubectl exec {{ pumba_pod.stdout }} -n {{ app_ns }}
+#    -- pumba --debug netem --interface eth0 -d {{ NETWORK_DELAY }}s loss
+#    -p 100 re2:^k8s_cstor-pool_{{ targeted_replica_pod }}
+#  args:
+#    executable: /bin/bash
+#  async: 400
+#  poll: 0
+#  register: async_results
+
+#- name: Wait for 60s post fault injection
+#  wait_for:
+#    timeout: 60
+
+- name: Generate database name
+  shell: cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1
+  register: new_db_name
+
+- name: Create new database in the mysql
+  include_tasks: "/common/utils/mysql_data_persistence.yml"
+  vars:
+    status: 'LOAD'
+    #ns: "app-percona-ns"
+    #pod_name: "percona-786c6b7c7f-7qkxm"
+    ns: "{{ app_ns }}"
+    pod_name: "{{ app_pod.stdout }}"
+    dbuser: 'root'
+    dbpassword: 'k8sDem0'
+    dbname: "{{ new_db_name.stdout }}"
+
+- name: Check if target got any new write IOs
+  shell: kubectl exec -it {{ istgt_replica.stdout }} -n openebs --container cstor-istgt -- istgtcontrol -q iostats |json_pp| grep WriteIOPS |awk -F ':' '{print $2}' | tr -d -c 0-9
+  register: iostats
+  retries: 3
+  delay: 3
+  until: iostats.stdout != last_iostats.stdout
+
+#- name: Check pumba async task status
+#  async_status:
+#    jid: "{{ async_results.ansible_job_id }}"
+#  register: async_poll_results
+#  until: async_poll_results.finished
+#  retries: 30
+#  delay: 15
+
+- name: Remove packet loss rule from targeted replica pod
```

Comment:
we may need to add check before this looking for `iostats.stdout != last_iostats.stdout`.. if they are not equal, it means IOs are written and need to fail the test?

Reply:
ok.. I think we need to make sure that iostats.stdout is not equal to last_iostats.stdout

Reply:
or increase the time

Reply:
we are checking for `last_iostats.stdout != iostats.stdout`